### PR TITLE
[MRG] FIX : plot_stat_map with MGZ anat

### LINKED ISSUE #577: pb with plot_stat_map with MGZ T1 image
we use a plot_stat_map in MNE:

http://martinos.org/mne/dev/auto_examples/inverse/plot_lcmv_beamformer_volume.html#example-inverse-plot-lcmv-beamformer-volume-py

with the current master of nilearn it fails due to the dtype (int8) of the anat I make available at:

https://dl.dropboxusercontent.com/u/2140486/T1.mgz

error is:

```
/Users/alex/work/src/nibabel/nibabel/freesurfer/mghformat.pyc in set_data_dtype(self, datatype)
    284             code = self._data_type_codes[datatype]
    285         except KeyError:
--> 286             raise MGHError('datatype dtype "%s" not recognized' % datatype)
    287         self._header_data['type'] = code
    288

MGHError: datatype dtype "int8" not recognized
```

any help?

### PR DESCRIPTION
**Pull request by agramfort**
closes #577

### CODE REVIEW

**Comment by agramfort** [comment]
not sure how to add a test for this... just test new_img_like with an MGZ image? is it worth adding an mgz image to the repo for this?

**Comment by GaelVaroquaux** [comment]
It's worth adding a small mgz to the repo to test this. Just make it small.

**Comment by Eric89GXL** [comment]
Travis complains on the `NIBABEL_VERSION="1.1.0"` run:
```
line 83, in new_img_like
    if isinstance(ref_img, nibabel.freesurfer.mghformat.MGHImage):
AttributeError: 'module' object has no attribute 'freesurfer'
```
So I guess you'll have to also modify for old `nibabel`.

**Comment by agramfort** [comment]
let's see if travis is happy...

**Comment by agramfort** [comment]
travis is happy

**Comment by agramfort** [comment]
ping @GaelVaroquaux

**Comment by AlexandreAbraham** [comment]
For the record, here is a small code to test accepted types in nibabel:

```python
import nibabel
import numpy as np


def _nibabel_image_types():
    for i in nibabel.__dict__:
        if i.endswith('Image'):
            yield nibabel.__dict__[i]


types_to_test = [
        np.int8,
        np.uint8,
        np.bool,
        np.float64,
]

for image in _nibabel_image_types():
    supp = []
    unsupp = []
    for t in types_to_test:
        try:
            image(np.zeros((10, 10, 10), dtype=t), np.eye(4))
            supp.append(t)
        except:
            unsupp.append(t)
    print('Type', image)
    print('\tSupported:', supp)
    print('\tUnsupported:', unsupp)
```

And here are the results:

```
Type <class 'nibabel.nifti1.Nifti1Image'>
        Supported: [<class 'numpy.int8'>, <class 'numpy.uint8'>, <class 'numpy.float64'>]
        Unsupported: [<class 'bool'>]
Type <class 'nibabel.minc2.Minc2Image'>
        Supported: [<class 'numpy.int8'>, <class 'numpy.uint8'>, <class 'bool'>, <class 'numpy.float64'>]
        Unsupported: []
Type <class 'nibabel.spm99analyze.Spm99AnalyzeImage'>
        Supported: [<class 'numpy.uint8'>, <class 'numpy.float64'>]
        Unsupported: [<class 'numpy.int8'>, <class 'bool'>]
Type <class 'nibabel.spm2analyze.Spm2AnalyzeImage'>
        Supported: [<class 'numpy.uint8'>, <class 'numpy.float64'>]
        Unsupported: [<class 'numpy.int8'>, <class 'bool'>]
Type <class 'nibabel.freesurfer.mghformat.MGHImage'>
        Supported: [<class 'numpy.uint8'>]
        Unsupported: [<class 'numpy.int8'>, <class 'bool'>, <class 'numpy.float64'>]
Type <class 'nibabel.minc1.MincImage'>
        Supported: [<class 'numpy.int8'>, <class 'numpy.uint8'>, <class 'bool'>, <class 'numpy.float64'>]
        Unsupported: []
Type <class 'nibabel.minc1.Minc1Image'>
        Supported: [<class 'numpy.int8'>, <class 'numpy.uint8'>, <class 'bool'>, <class 'numpy.float64'>]
        Unsupported: []
Type <class 'nibabel.analyze.AnalyzeImage'>
        Supported: [<class 'numpy.uint8'>, <class 'numpy.float64'>]
        Unsupported: [<class 'numpy.int8'>, <class 'bool'>]
Type <class 'nibabel.nifti2.Nifti2Image'>
        Supported: [<class 'numpy.int8'>, <class 'numpy.uint8'>, <class 'numpy.float64'>]
        Unsupported: [<class 'bool'>]
```

So, 2 remarks:

* we should extend the condition to Spm*AnalyzeImage because they don't support int8 either
* when we use internal masks (90% of usecases if masks) we could rely on MincImage format since it supports bool. This would be easier than passing the data and affine separately in all the functions.

**Comment by agramfort** [comment]
this avoids adding a file to the repo. Do you want to take over my PR?

**Comment by AlexandreAbraham** [comment]
No, I think that your PR should be merged to solve your problem. Supporting all other image types is a bigger task and would take more time. I just wanted to raise this since you're talking about image types.

:+1: for merge and creating an issue about other image types.

**Comment by agramfort** [comment]
fine with me

@GaelVaroquaux any objection?

File: nilearn/tests/test_niimg.py [hunk 1]
```diff
@@ -1,7 +1,28 @@
+import os
+from distutils.version import LooseVersion
+import numpy as np
+import nibabel as nib
```
Comment (GaelVaroquaux) [inline nilearn/tests/test_niimg.py]: Please don't rename nibabel. It's not worth saving those 4 letters.

Comment (agramfort) [inline nilearn/tests/test_niimg.py]: done

File: nilearn/tests/test_niimg.py [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+datadir = os.path.join(currdir, 'data')
+
 
 def test_copy_img():
     assert_raises_regex(ValueError, "Input value is not an image",
-        niimg.copy_img, 3)
+                        niimg.copy_img, 3)
+
+
+def test_new_img_like_mgz():
```
Comment (GaelVaroquaux) [inline nilearn/tests/test_niimg.py]: Please tell us (in a one-line comment) why this test is useful.

**Comment by GaelVaroquaux** [comment]
Thank you. Merging

**Comment by agramfort** [comment]
thanks !

**Pull request merged**